### PR TITLE
Make mongod port configurable

### PR DESCRIPTION
**Pull request by mircea-cosbuc**
This addresses #839 and moves `mongod` configurations handled by the operator to the `MongodConfiguration` type.

### All Submissions:

* [ ] Have you opened an Issue before filing this PR?
* [ ] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [ ] Have you signed all of your commits?
* [ ] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [ ] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).

### CODE REVIEW

File: pkg/automationconfig/automation_config_builder.go
Comment (irajdeep) [inline pkg/automationconfig/automation_config_builder.go:20]: The `defaultPort` is also declared at `mongodbcommunity_types.go`, is it possible to consolidate it to only one place?

File: api/v1/mongodbcommunity_types.go
Comment (irajdeep) [inline api/v1/mongodbcommunity_types.go:298]:
```suggestion
// If port is not specified, the default port of 27017 will be used.
```

File: controllers/construct/build_statefulset_test.go
Comment (mircea-cosbuc) [reply controllers/construct/build_statefulset_test.go:81]: Yes, that test covers it. We could break that test into multiple ones if we have more `mongod` configuration options we want to handle.

File: pkg/automationconfig/automation_config_builder.go
Comment (irajdeep) [reply pkg/automationconfig/automation_config_builder.go:20]: We have been trying to avoid using `util` package for a while now, basically, we are moving out of the util package in enterprise since it's very unclear what those collections of packages entail. 

I still this this should be declared at one place to avoid any potential bug later, but will leave it to you to decided the preferred location.

Comment (irajdeep) [reply pkg/automationconfig/automation_config_builder.go:20]: Just noticed we already import `automationconfig` package in `mongodbcommunity_types.go`, so maybe we can reuse this constant in mongodbcommunity_types.go ?

File: controllers/construct/mongodbstatefulset.go
Comment (chatton) [inline controllers/construct/mongodbstatefulset.go:89]: this interface is used in enterprise (for deploying the AppDB). I think this will still be fine, but next time we update the community operator version library in the enterprise repo we can address this.

**Review by chatton — APPROVED** [review]
Reviewed together in person, LGTM! 

Thanks for moving some of those. values on the `MongodConfiguration`, I think that makes a lot more sense.

I think this is good to merge once @irajdeep's comments are resolved.

**Review by irajdeep — APPROVED** [review]
LGTM 👍

File: controllers/construct/mongodbstatefulset.go
Comment (mircea-cosbuc) [reply controllers/construct/mongodbstatefulset.go:89]: It might also be a good reason to synchronise the way we're handling the `mongod` options in the two operators.

**Pull request merged**
